GreysonElkins/exquisite-corpse
# Feature/ge api to bookshelf

#### What's this PR do?
* Merges @nickhartdev's PR with Greyson's worked fixed up
  * Refactors StoryEdit to be a functional component
  * Moves all state info and relevant methods to StoryEditView
  * Refactors the fetch call to use getData instead of getRandomPrompt
* Greyson's additions: 
  * App holds stories(updated periodically to reflect actual prompts and authors)
  * App holds authors & prompts _(realizing now that some of this data is technically duplicated and used in stories, don't know if that's kosher, but it's definitely getting used)_
  * Bookshelves and Books are responsible for cleaning story data
  * Redirect from an unfinished book doesn't require additional fetching
#### Where should the reviewer start?
**Nick:** StorySetupView
**Greyson:** Bookshelves in Library and Welcome views
#### How should this be manually tested?
**Nick** Click new story, choose a prompt, and start your story! A prompt from the appropriate genre should render if the user chooses to use one.
**Greyson** I broke Nick's thing but Imma fix it. 
#### Any background context?
I'll revisit this to take care of ticket #35. The original plan was to write methods to utilize the fact that App holds all prompts, but I ran out of time so the original fetch call is still there.
#### What are the relevant tickets?
None yet!
#### Screenshots (if appropriate)
![cat](https://user-images.githubusercontent.com/60753853/93148007-64931e80-f6b8-11ea-8281-98af69d9c711.gif)
